### DISCUSSION TITLE
Support for EC300?

### DISCUSSION THREAD

**Discussion by tcurdt** [grblHAL/Controllers#11] [Q&A]
Remora supports the EC300 style boards. https://www.nvcnc.net/ec300.html
Does that mean there is a chance to get gbrlHAL working on there?

**Comment by terjeio** [comment]
It uses a iMXRT1052 processor so a driver has to be written for it. I do not know how different it is from iMXRT1062, it might be possible to reuse part of that driver - but since the Teensy 4.1 Arduino framework is referenced by it that may have to be modified as well.

**Reply by tcurdt** [reply]
I think there is also a stm32 version.

https://github.com/scottalford75/Remora-EC500

And for sure the NVEM (a very very similar board) is stm32

https://github.com/scottalford75/Remora-NVEM

**Reply by terjeio** [reply]
Both are out of production?
The EC500 is F407 based so only a board map is required?
The NVN is F402 based so will need a board map and code to set up the clock tree.

**Reply by tcurdt** [reply]
It sure says so here https://remora-docs.readthedocs.io/en/latest/hardware/Ethernet/ethernet.html#controller-boards
Tough to say with these Chinese boards - but I assume Scott knows.

There are still plenty available though.
Hence I was wondering if it was possible to get it working.
Sounds like that could be doable with reasonable effort - at least for the older stm ones.

It certainly sounds like RT1052 support would be something that needs looking into to support these boards further.
Seems like this also supports the RT1052:

https://github.com/Seeed-Studio/ArduinoCore-imxrt

I guess it depends how eager one is - but it sounds doable in theory :)
